golang/pkg/signaling/server.go

```diff
@@ -24,6 +24,11 @@ func NewServer(room *Room) *Server {
 }
 
 func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	upgrader.CheckOrigin = func(r *http.Request) bool {
```

Comment:
This would set it every time someone connects.

Reply:
This needs to be in NewServer. Also move the creation of upgrader into NewServer too and store it in a pointer on the Server struct.